go/vt/mysqlctl/builtinbackupengine.go

```diff
@@ -882,16 +887,20 @@ func (be *BuiltinBackupEngine) restoreFiles(ctx context.Context, params RestoreP
 		wg.Add(1)
 		go func(i int) {
 			defer wg.Done()
-
-			// Wait until we are ready to go, skip if we already
-			// encountered an error.
-			sema.Acquire(ctx, 1)
+			fe := &fes[i]
+			// Wait until we are ready to go, return if we encounter an error
+			acqErr := sema.Acquire(ctx, 1)
+			if acqErr != nil {
+				log.Errorf("not able to acquire semaphore to restore file:%s  err: %s", fe.Name, acqErr.Error())
+				rec.RecordError(acqErr)
+				return
+			}
```

Comment:
Same not here about checking for context cancellation.

Reply:
done